server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -152,14 +155,16 @@ public void startReplication(ActionListener<Void> listener) {
 
     private void getFiles(CheckpointInfoResponse checkpointInfo, StepListener<GetSegmentFilesResponse> getFilesListener)
         throws IOException {
+        state.setStage(SegmentReplicationState.Stage.FILE_DIFF);
         final Store.MetadataSnapshot snapshot = checkpointInfo.getSnapshot();
         Store.MetadataSnapshot localMetadata = getMetadataSnapshot();
         final Store.RecoveryDiff diff = snapshot.segmentReplicationDiff(localMetadata);
-        logger.debug("Replication diff {}", diff);
-        // Segments are immutable. So if the replica has any segments with the same name that differ from the one in the incoming snapshot
-        // from
-        // source that means the local copy of the segment has been corrupted/changed in some way and we throw an IllegalStateException to
-        // fail the shard
+        logger.trace("Replication diff {}", diff);
```

Comment:
Shouldn't it be `ParameterizedMessage` ?

Reply:
@reta IMO `ParameterizedMessage` isn't needed here for two reasons:

1. Log4j's logging APIs [directly support the parameterized message format](https://logging.apache.org/log4j/log4j-2.12.4/manual/api.html) 
2. We're not using lazy logging here since the objects being dumped don't involve heavyweight computations

That said, I don't have a strong opinion either way so let me know if you'd recommend using lazy logging here.